app/models/concerns/variant_stock.rb

```diff
@@ -100,7 +100,7 @@ def can_supply?(quantity)
   # Here we depend only on variant.total_on_hand and variant.on_demand.
   #   This way, variant_overrides only need to override variant.total_on_hand and variant.on_demand.
   def fill_status(quantity)
-    if on_hand >= quantity
+    if on_hand >= quantity || on_demand
```

Comment:
do you know if we could we just get rid of backordered here and always return on_hand = quantity and backordered = 0?
now that I consider this, maybe we never get to backordered if variant is on_hand, that would mean we are fullfilling an order with outofstock variant.

so, this change may make, the else part of this if, dead code... what do you think?